src/framework/components/script/component.js

```diff
@@ -363,8 +387,43 @@ class ScriptComponent extends Component {
     }
 
     _onInitializeAttributes() {
-        for (let i = 0, len = this.scripts.length; i < len; i++)
-            this.scripts[i].__initializeAttributes();
+        this.initializeAllScriptAttributes();
+    }
+
+    initializeAllScriptAttributes() {
+
+        for (let i = 0, len = this.scripts.length; i < len; i++) {
+            const script = this.scripts[i];
+            this.initializeAttributes(script);
+        }
+    }
+
+    initializeAttributes(script) {
+
+        // if script has __initializeAttributes method assume it has a runtime schema
+        if (script instanceof ScriptType) {
+
+            script.__initializeAttributes();
+
+        } else {
+
+            const toLowerCamelCase = str => str[0].toLowerCase() + str.substring(1);
+
+            // otherwise we need to manually initialize attributes from the schema
+            const name = script.__scriptType.__name || toLowerCamelCase(getScriptName(script.__scriptType));
```

Comment:
it feels like this should be a function as this is in the code base multiple times.